crates/cxx-qt-lib/src/core/qtypes.rs

```diff
@@ -0,0 +1,253 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::ops::{Deref, DerefMut};
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qtypes.h");
+    }
+}
+
+/// Typedef for long long int. This type is guaranteed to be 64-bit on all platforms supported by Qt.
+#[repr(transparent)]
+#[derive(Default, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
I would add `Clone` and `Copy` to this list as well...

Reply:
done